contrib/parseq-examples/src/main/java/com/linkedin/parseq/example/batching/BatchingClientExample.java

```diff
@@ -0,0 +1,70 @@
+/* $Id$ */
+package com.linkedin.parseq.example.batching;
+
+import java.util.List;
+
+import com.linkedin.parseq.Engine;
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.batching.BatchEntry;
+import com.linkedin.parseq.batching.BatchingStrategy;
+import com.linkedin.parseq.batching.BatchingSupport;
+import com.linkedin.parseq.example.common.AbstractExample;
+import com.linkedin.parseq.example.common.ExampleUtil;
+import com.linkedin.parseq.promise.Promises;
+
+
+/**
+ * @author Jaroslaw Odzga (jodzga@linkedin.com)
+ */
+public class BatchingClientExample extends AbstractExample {
+
+  public static class ExampleBatchingStrategy extends BatchingStrategy<Long, Long> {
+
+    @Override
+    public Long groupForKey(Long key) {
+      //lets say all tasks can be grouped into one batch
+      return 0L;
+    }
+
+    @Override
+    public Task<?> taskForBatch(final Long group, final List<? extends BatchEntry<Long, Object>> batch) {
```

Comment:
This is tricky. Not really sure how this will play out. Will need to prototype it for some types of tasks e.g. rest.li.
The reason for downcasting everything to Object is that you may want to be able to batch together tasks that don't have any type relationships. For example, if we used multiplexed reat.li requests we might be able to batch together not only GET operations but really arbitrary operations.
I think it is necessary to prototype few implementations first to see how this API would play out.